lib/src/main/kotlin/app/cash/kfsm/v011/MachineBuilder.kt

```diff
@@ -34,11 +34,15 @@ class MachineBuilder<ID, V : Value<ID, V, S>, S : State<ID, V, S>> {
      *
      * @param effect The effect to apply during the transition
      * @throws IllegalStateException if a transition to this state is already defined
+     * @throws IllegalStateException if the from state cannot transition to this state
      */
     infix fun S.via(effect: Effect<ID, V, S>) {
       if (this@via in transitions) {
         throw IllegalStateException("State $this already has a transition defined from $from")
       }
+      if (!from.canDirectlyTransitionTo(this@via)) {
+        throw IllegalStateException("State $from cannot transition to $this")
+      }
```

Comment:
It feels like this starts to conflate the definitions of transitions, with transitions being defined in the machine and in the transition itself.

Perhaps this error message could be more specific:
```
State $from cannot transition to $this. The transition's "to" state may be misconfigured.
```